crates/vm/src/arch/segment.rs

```diff
@@ -60,11 +53,11 @@ impl<F: PrimeField32, VC: VmConfig<F>> ExecutionSegment<F, VC> {
         program: Program<F>,
         init_streams: Streams<F>,
         initial_memory: Option<Equipartition<F, CHUNK>>,
-        fn_bounds: FnBounds,
+        _fn_bounds: FnBounds,
```

Comment:
This is a bit weird since it's used for `bench-metrics` feature. Maybe just silence the clippy warning instead?